src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
Personally, i'd prefer we just do the work to get this correct now.

Race conditions and complex interaction models are really bad for the health of our code.  They lead to instability and terrifically difficult to track down problems.  We should push for simple abstractions for this sort of thing that both isolate the complexity and make it very difficult for downstream clients to get things wrong.  This helps avoid whole classes of bugs, and it makes it possible to focus correctness on a single part of the system.

Since we're already working to eliminate the raceyness here, i'd much rather we just do it now instead of potentially missing something and having to deal with this same issue a month/year down the line.